rholang/examples/linking/packages/SignatureCondition.rho

```diff
@@ -0,0 +1,13 @@
+export SignatureCondition in {
+  import VerifySignature in {
+    contract SignatureCondition(publicKey, return) = {
+      new condition in {
+        //evidence = channel where data can be sent for signature verification
+        contract condition(evidence, return) = {
+          VerifySignature(publicKey, evidence, return)
```

Comment:
Sending random data on the evidence channel and waiting for a signature result to come back is possibly too interactive. Perhaps instead we should simply have a transaction counter here similar to Ethereum and the evidence would then be `[txnNumber, signature]` where we would require the given `txnNumber` to match the internal counter and `signature` would be verified through a call to `ConfirmResult(publicKey, txnNumber, signature, return)` (see `VerifySignature.rho`). If all that succeeds then we increment the internal transaction counter. This model allows an honest user to directly initiate a transaction instead of having to initiate it, wait for the verification data, and send signed data back.

Would be interested to know if people have opinions either way.